docs/testing/webdriverio/02-unit-testing.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Unit Testing
+sidebar_label: Unit Testing
+description: Unit Testing
+slug: /testing/webdriverio/unit-testing
+---
+
+# Unit Testing
+
+WebdriverIO makes it easy to unit test components and app utility functions in the browser. Unit tests validate the code in isolation. Well written tests are fast, repeatable, and easy to reason about. It tries to follow a simple guiding principle: the more your tests resemble the way your software is used, the more confidence they can give you.
+
+### Test Setup
+
+To resemble how your component is being used as close as possible to reality we need to render it into an actual DOM tree. WebdriverIO provides a helper package for this that you can use called `@wdio/browser-runner/stencil`. It exports a `render ` method that allows us to mount our component to the DOM.
+
+For example, given the following component:
+
+```ts reference title="/src/components/my-component/my-component.tsx"
+https://github.com/webdriverio/component-testing-examples/blob/main/stencil-component-starter/src/components/my-component/my-component.tsx
+```
+
+We import the component into our test to render it in the browser:
+
+```ts reference title="/src/components/my-component/my-component.test.tsx"
+https://github.com/webdriverio/component-testing-examples/blob/main/stencil-component-starter/src/components/my-component/my-component.test.tsx#L2-L18
+```
+
+You can find more information about the `render` methods option and its return object in the WebdriverIO [documentation](https://webdriver.io/docs/component-testing/stencil#render-options).
+
+### Handle Asynchronicity
+
+Instead of directly working on DOM objects, with WebdriverIO you are interacting with references of DOM nodes and interact through WebDriver commands that are async. Make sure you always use an `await` to ensure that all commands and assertion are executed sequentially.
+
+:::info
+
+Missing an `await` can be a simple oversight and can cause us long hours of debugging. To avoid this and ensure promises are handled properly, it is recommended to use an ESLint rule called [`require-await`](https://eslint.org/docs/latest/rules/require-await).
```

Comment:
Ooh nice, we'll wanna recommend this in Playwright too. Sean said Framework has a lint rule, but idk if they used a published rule or rolled their own.